File: dotnet/GANP.Website/Controllers/HomeController.cs
using System.Web.Mvc;
using GANP.Website.Models;
using GANP.Website.Services;

namespace GANP.Website.Controllers
{
    public class HomeController : Controller
    {
        private IFileOperationService _reader;
        private IGeneratePasswordService _service;
        public HomeController()
        {
            _reader = new FileOperationService();
            _service = new GeneratePasswordService(_reader,new HttpFilePathService());
        }

        public HomeController(IFileOperationService fileService, IGeneratePasswordService service)
        {
            _reader = fileService;
            _service = service;
        }

        public ActionResult Index()
        {
            var options = new GeneratePasswordOptions();
            var newPassword = _service.GeneratePassword(options);
            var model = new GeneratePasswordModel(newPassword, options);
            return View(model);
        }
    }
}

File: java/src/com/getanewpassword/controllers/HomeController.java
package com.getanewpassword.controllers;

import com.getanewpassword.models.GeneratePasswordOptions;
import com.getanewpassword.services.GeneratePasswordService;
import com.getanewpassword.services.GeneratePasswordServiceImpl;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.context.ServletContextAware;
import org.springframework.web.servlet.ModelAndView;

import javax.servlet.ServletContext;
import java.io.IOException;

@Controller
public class HomeController  implements ServletContextAware {
    String wordFilePath;

    @RequestMapping("/")
    public ModelAndView index() throws IOException {
        GeneratePasswordService svc = new GeneratePasswordServiceImpl(wordFilePath);
        String password = svc.GeneratePassword(new GeneratePasswordOptions());

        return new ModelAndView("index", "newPassword", password);
    }

    @RequestMapping(value = "/api/generatepassword", method = RequestMethod.GET, produces = "application/json")
    @ResponseBody
    public String generatePassword() throws IOException {
        GeneratePasswordService svc = new GeneratePasswordServiceImpl(wordFilePath);
        String password = svc.GeneratePassword(new GeneratePasswordOptions());
        return "{\"Password\" : \"" + password + "\"}";
    }

    @Override
    public void setServletContext(ServletContext servletCtx){
        wordFilePath = servletCtx.getRealPath("content/wordlist.txt");
    }
}